### PR TITLE
roachtest: improve end-user documentation

### PR DESCRIPTION
**Pull request by tbg**
This (attempts) to give some practical tips on how to get to the
point of invoking a simple roachtest, and provides troubleshooting
advice.

I also added a developer-centric section which covers (at a very high
level) adding new tests and reproducing test failures.

cc @vy-ton

Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/README.md [hunk 1]
```diff
@@ -6,38 +6,169 @@ separate) tool `roachprod`.
 
 # Setup
 
-1. [Set up `roachprod`](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachprod/README.md), if you haven't already. This includes making sure `$PWD/bin` is on your `PATH` and `gcloud` is installed and properly configured.
-1. Build a linux release binary of `cockroach`: `build/builder.sh mkrelease amd64-linux-gnu`
-1. Build a linux binary of the `workload` tool: `build/builder.sh mkrelease amd64-linux-gnu bin/workload`
-1. Build a local binary of `roachtest`: `make bin/roachtest`
+- [Set up `roachprod`](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachprod/README.md), if you haven't already.
+- If you want to run tests that require an Enterprise license, add `COCKROACH_DEV_LICENSE=<key>` to your path. Cockroach Labs employees can find one in [Slack](https://cockroachlabs.slack.com/archives/CJX8V9SJ2/p1597348203368800?thread_ts=1597348076.367800&cid=CJX8V9SJ2).
+- For Cockroach Labs employees, if your local username does not match your
```
Comment (rickystewart) [inline pkg/cmd/roachtest/README.md:11]: Love this, thanks :)

**Review by rickystewart — APPROVED** [review]
stamp

File: pkg/cmd/roachtest/README.md [hunk 2]
```diff
@@ -6,38 +6,169 @@ separate) tool `roachprod`.
...
+
+The `cockroach` and `workload` binaries need to be built for the architecture
+of the target cluster. In most cases, this means `amd64-linux-gnu`, i.e. OSX
+users have to either download the binary they need from somewhere (see below)
+or cross-compile the binary locally, which unfortunately takes a long time (due
+to the overhead of virtualization for Docker on Mac).
+
+As a general rule of thumb, if running against "real" VMs, and the version to
+test is checked out, the following will build the right binaries and put them
+where `roachtest` will find them:
```
Comment (erikgrinaker) [inline pkg/cmd/roachtest/README.md:55]: Might be worth mentioning somewhere around here that on macOS, `roachtest` will prefer `bin/workload` if present (a macOS binary) and fall back to `bin.docker_amd64/workload` (the cross-compiled Linux binary) if it does not exist. This is kind of non-discoverable, and tripped me up a couple of times when I first started out with `roachtest` -- macOS users should either make sure `bin/workload` does not exist, or pass the path explicitly.

**Comment by tbg** [comment]
TODO: incorporate https://github.com/cockroachdb/cockroach/pull/67913#issuecomment-884845624

File: pkg/cmd/roachtest/README.md (hunk 2 above)
Comment (tbg) [reply pkg/cmd/roachtest/README.md:55]: Thanks, I added a note.

Comment (erikgrinaker) [reply pkg/cmd/roachtest/README.md:55]: Did you forget to push?

Comment (tbg) [reply pkg/cmd/roachtest/README.md:55]: No, still working on the PR. Sorry, should've staged that comment. Still getting used to using vanilla Github reviews more :-)
I'll ping when it's ready.

Comment (tbg) [reply pkg/cmd/roachtest/README.md:55]: Updated.

**Review by erikgrinaker — APPROVED** [review]

**Comment by tbg** [comment]
bors r=erikgrinaker

**Pull request merged**